src/v/cloud_storage/remote_partition.cc

```diff
@@ -190,16 +191,19 @@ class partition_record_batch_reader_impl final
               "gate_closed_exception while reading from remote_partition");
             _it = _end;
             _reader = {};
-        } catch (const std::exception& e) {
+        } catch (const std::bad_alloc& e) {
             vlog(
               _ctxlog.warn,
-              "exception thrown while reading from remote_partition: {}",
+              "bad_alloc thrown while reading from remote_partition: {}",
               e.what());
-            _it = _end;
-            _reader = {};
+            bad_alloc_exception_ptr = std::current_exception();
+        }
 
-            throw;
+        if (bad_alloc_exception_ptr) {
+            co_await set_end_of_stream();
+            std::rethrow_exception(bad_alloc_exception_ptr);
```

Comment:
i don't know enough about what is going on here to really judge, but i'm having trouble understanding what is happening. the title says we are stopping the reader on exception, but it looks like we are only stopping it on bad_alloc. on the other hand, in the previous version, any exception appeared to reset the reader (but not stop it), and now that isn't done for any exception.

maybe a few examples or a link to an issue with more detail would be helpful. i'm also open to the idea that i have no idea what i'm talking about and shouldn't comment :)

Reply:
You're right. The code and commit message got out sync. I've updated both now and hopefully it makes sense.